src/Columns/ColumnArray.cpp

```diff
@@ -533,7 +533,7 @@ void ColumnArray::insertRangeFrom(const IColumn & src, size_t start, size_t leng
 
     Offsets & cur_offsets = getOffsets();
     /// Reserve offsets before to make it more exception safe (in case of MEMORY_LIMIT_EXCEEDED)
-    cur_offsets.reserve(cur_offsets.size() + length);
```

Comment:
I think it makes sense to keep `reserve` in `insertRangeFrom` because there are a lot of places where it is used in al loop with relatively small lengths (e.g. [here](https://github.com/ClickHouse/ClickHouse/blob/31479649fe16c507265870ef5635cb4e5cc8bc3b/src/Processors/Transforms/SortingTransform.cpp#L118) and [here](https://github.com/ClickHouse/ClickHouse/blob/31479649fe16c507265870ef5635cb4e5cc8bc3b/src/Processors/Transforms/ColumnGathererTransform.h#L194)) which will lead to large number of `realloc` calls.

Reply:
Thank you! I wanted to avoid these cases but didn't pay attention...